### PR TITLE
Introduce content folder for hacky blog posts and snippets.

### PR DESCRIPTION
**Pull request by klizhentas**

### CODE REVIEW

**Comment by webvictim** [comment]
IMO this should be kept separately to the Teleport repo. People neither need nor want to clone non-code when checking out Teleport.

**Comment by klizhentas** [comment]
We already have docs, examples and other snippets here, so people can survive a couple of md files :) I want the process to be as easy as possible for developers to submit content, so we will give this a try.

**Comment by markherring** [comment]
Would it be better to rather have this in the gravitational/web repository? Still do PRs all the same but it seems more related to the web than the product?

**Comment by klizhentas** [comment]
@markherring let's try it here, it makes sense to have Teleport focused content in teleport repository.

**Comment by webvictim** [comment]
I personally have a preference for keeping it outside the repo, but I do also understand Sasha's logic that it's easier for engineers and outside contributors to propose Teleport-related content when the folder is inside the repo itself. I can live with it if everyone's on board.

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:20]:
```suggestion
## Language Runtimes and Multi-tenancy
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:23]:
```suggestion
Let’s take a simple Go “SaaS” service that fetches the contents of a URL
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:46]: Should the `defer` should be before the error handler?

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:82]:
```suggestion
sleepingGoroutines.Add(goroutine0)
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:87]:
```suggestion
sleepingGoroutines.Add(goroutine1)
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:101]:
```suggestion
combination of using the asynchronous [epoll](https://kovyrin.net/2006/04/13/epoll-asynchronous-network-programming/) Linux
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:102]:
```suggestion
function that can efficiently poll many handles at once, and the powerful Go runtime
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:103]:
```suggestion
that handles `go` function calls using a smart scheduler.
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:111]:
```suggestion
So what’s the problem? Well, the Go scheduler (and other similar schedulers) do not really handle
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:112]:
```suggestion
situations like this one well:
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:157]:
```suggestion
In this case,  you would not observe the same parallelism. In my case, the function
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:158]:
```suggestion
will print something like this (let’s ignore the problems with the factorial implementation for a moment):
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:175]:
```suggestion
and number crunching, tenant 0 really consumed all the CPU time and degraded performance for all other tenants.
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:179]:
```suggestion
suffered an outage due to a bug in the code for all our customers, PagerDuty is ringing and we are
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:184]:
```suggestion
The core of the problem is that CPU time is limited, and the programming language runtime is not
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:185]:
```suggestion
capable of efficiently distributing the CPU time per tenant. It lacks context and control over the
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:188]:
```suggestion
In a single tenant world, this is still a problem - but by assuming that only one tenant
```

File: content/blog/multitenancy.md
Comment (webvictim) [inline content/blog/multitenancy.md:189]:
```suggestion
uses a single process at a time, we have more control over the isolation:
```

File: .github/CODEOWNERS
Comment (webvictim) [inline .github/CODEOWNERS:20]: I'm willing to review this stuff as well if you want.

```suggestion
/content/ @russjones @markherring @benarent @gregchase @webvictim
```

**Review by webvictim — COMMENTED** [review]
Mostly spelling/formatting/grammar

**Review by benarent — APPROVED** [review]

**Comment by benarent** [comment]
I enjoyed the blog post, and loved that it was short, technical and straight to the point. I think for now, we don't have to worry about markdown files are going to take up too much repo space... ( images and videos for docs.. might be worth moving out soon since it's starting to bloat the git repo )

**Comment by webvictim** [comment]
Yes - my thinking was ultimately that if we have the sources for posts here, it's a natural progression that we may soon afterward need to start hosting images and videos here too - I think if it comes to that, we need to move it elsewhere. This repo is cloned in full hundreds of times a day.

**Comment by markherring** [comment]
how do I know when multi-tenancy has been "blessed" by you all?

**Comment by webvictim** [comment]
I requested some grammatical fixes - once @klizhentas has done those I'll approve the PR.

**Comment by markherring** [comment]
Are you the final approver or is it all reviewers?

**Comment by webvictim** [comment]
I would go with all reviewers. Be nice for @russjones to take a look too.

**Pull request closed without merging**
